buildvertical.targets

```diff
@@ -27,13 +27,21 @@
     </MSBuild>
 
     <ItemGroup>
-      <!-- transform back to project -->
-      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')">
-        <AdditionalProperties>Configuration=%(Identity);%(_ProjectBestConfigurations.AdditionalProperties)</AdditionalProperties>
+      <!-- transform back to project keeping the original item order. -->
+      <_ProjectBestConfigurations>
+        <Configuration>%(Identity)</Configuration>
+      </_ProjectBestConfigurations>
+      <ProjectWithConfiguration Include="@(_ProjectBestConfigurations->'%(OriginalItemSpec)')" />
+      <ProjectWithConfiguration>
+        <AdditionalProperties>Configuration=%(Configuration);%(ProjectWithConfiguration.AdditionalProperties)</AdditionalProperties>
       </ProjectWithConfiguration>
     </ItemGroup>
 
     <ItemGroup>
+      <ProjectWithConfiguration RemoveMetadata="Configuration" />
```

Comment:
Nevermind I found https://msdn.microsoft.com/en-us/library/ms171453.aspx#BKMK_RemoveMetadata. However my second question still stands why are you removing Configuration?